lib/usd/translators/meshWriter_BlendShapes.cpp

```diff
@@ -0,0 +1,977 @@
+            deformedMeshDagPath.fullPathName().asChar());
+        return MObject::kNullObj;
+    }
+
+    size_t numOfBlendShapeDeformers = blendShapeDeformerInfos.size();
+    switch (numOfBlendShapeDeformers) {
+    case 0:
+        TF_WARN(
+            "Cannot find any blendshape deformers for the mesh: %s",
+            deformedMeshDagPath.fullPathName().asChar());
+        return MObject::kNullObj;
+    case 1: break;
+    default:
+        // TODO: (yliangsiew) For multiple blend shape deformers, what do we do?
+        // Do we collapse the shapes from multiple blend targets together, or just
+        // write out only the "closest" blendshape deformer's targets? Or just write all
+        // of them and print this warning to end-users?
+        TF_WARN("Multiple blendshape deformers were found; while your shapes will still be saved, "
+                "since USDSkelBlendShape does not support a deformation stack, results may be "
+                "unpredictable on import.");
+        break;
+    }
+
+    bool exportAnim = !(exportArgs.timeSamples.empty());
+
+    SdfPathVector  usdBlendShapePaths;
+    VtTokenArray   usdBlendShapeNames;
+    const SdfPath& primSchemaPath = primSchema.GetPrim().GetPath();
+    for (size_t i = 0; i < numOfBlendShapeDeformers; ++i) {
+        MayaBlendShapeDatum blendShapeInfo = blendShapeDeformerInfos[i];
+        // NOTE: (yliangsiew) Each of the weights here we iterate over is equivalent
+        // to each individual weight that you are able to toggle on a blendshape node
+        // in the Attribute Editor within Maya.
+        for (unsigned int j = 0; j < blendShapeInfo.numWeights; ++j) {
+            MayaBlendShapeWeightDatum weightInfo = blendShapeInfo.weightDatas[j];
+            unsigned int              numTargetItemIndices = weightInfo.targetItemIndices.length();
+            switch (numTargetItemIndices) {
+            case 0:
+                TF_RUNTIME_ERROR("No target indices for the blendshape target could be found. "
+                                 "Check that the blendshape was set up correctly.");
+                return MObject::kNullObj;
+            case 1: { // NOTE: (yliangsiew) Means no inbetweens possible. i.e. [6000] only.
+                size_t numOfTargets = weightInfo.targets.size();
+                for (size_t k = 0; k < numOfTargets; ++k) {
+                    MayaBlendShapeTargetDatum targetDatum = weightInfo.targets[k];
+                    MObject                   targetMesh = targetDatum.targetMesh;
+                    MString                   curTargetNameMStr;
+                    if (!targetMesh.isNull()) {
+                        // NOTE: (yliangsiew) Because UsdSkelBlendShape does not
+                        // support animated targets (the `normalOffsets` and
+                        // `offsets` attributes are defined as uniforms), we cannot
+                        // fully support it in the exporter either.
```

Comment:
Ah that is an interesting limitation. We'd need UsdSkelBlendShape targets to accept the same data type that the input & output geometry are to have a more general solution where we can connect a graph of computes together.